code/Player/Player.PropPossesion.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Sandbox;
+
+namespace TTT;
+
+public partial class Player
+{
+	[Net]
+	public Prop Prop { get; private set; }
+
+	public void SimulatePossession()
+	{
+		if ( Input.Pressed( InputButton.Duck ) || !Prop.IsValid() )
+		{
+			CancelPossession();
+			return;
+		}
+
+		if ( Input.Pressed( InputButton.Jump ) || Input.Forward != 0f || Input.Left != 0f )
+			Prop.Components.Get<PropPossession>().Punch();
+	}
+
+	[Event.Entity.PreCleanup]
+	protected void CancelPossession()
+	{
+		if ( Prop.IsValid() )
+		{
+			Prop.Components.RemoveAny<PropPossession>();
+			Prop.Owner = null;
```

Comment:
Can we call `Prop.Owner = null` inside of `Deactivate()` of the `PropPossession` component?